src/sas/sascalc/fit/fitstate.py

```diff
@@ -0,0 +1,408 @@
+                                 % model['fit_page_source'])
+
+
+def model_name(state):
+    """
+    Build the model name out of form factor and structure factor (if present).
+
+    This will be the name that is stored as the model name in the simultaneous
+    fit model_list structure corresponding to the form factor and structure
+    factor given on the individual fit pages.  The model name is used to help
+    disambiguate different SASentry sections with the same dataset.
+    """
+    p_model, s_model = state.formfactorcombobox, state.structurecombobox
+    if s_model is not None and s_model != "" and s_model.lower() != "none":
+        return '*'.join((p_model, s_model))
+    else:
+        return p_model
+
+def get_data_weight(state):
+    """
+    Get error bars on data.  These could be the values computed by reduction
+    and stored in the file, the square root of the intensity (if instensity
+    is approximately counts), the intensity itself (would be better as a
+    percentage of the intensity, such as 2% or 5% depending on relative
+    counting time), or one for equal weight uncertainty depending on the
+    value of state.dI_*.
+    """
+    # Cribbed from perspectives/fitting/utils.py:get_weight and
+    # perspectives/fitting/fitpage.py: get_weight_flag
+    weight = None
+    if state.enable2D:
+        dy_data = state.data.err_data
+        data = state.data.data
+    else:
+        dy_data = state.data.dy
+        data = state.data.y
+    if state.dI_noweight:
+        weight = np.ones_like(data)
+    elif state.dI_didata:
+        weight = dy_data
+    elif state.dI_sqridata:
+        weight = np.sqrt(np.abs(data))
+    elif state.dI_idata:
+        weight = np.abs(data)
+    return weight
+
+_MODEL_CACHE = {}
+def load_model(name):
+    """
+    Given a model name load the Sasview shim model from sasmodels.
+
+    If name starts with "[Plug-in]" then load it as a custom model from the
+    plugins directory.  This code does not go through the Sasview model manager
+    interface since that loads all available models rather than just those
+    needed.
+    """
+    # Remember the models that are loaded so they are only loaded once.  While
+    # not strictly necessary (the models will use identical but different model
+    # info structure) it saves a little time and memory for the usual case
+    # where models are reused for simultaneous and batch fitting.
+    if name in _MODEL_CACHE:
+        return _MODEL_CACHE[name]
+    if name.startswith(PLUGIN_NAME_BASE):
+        name = name[len(PLUGIN_NAME_BASE):]
+        plugins_dir = find_plugins_dir()
+        path = os.path.abspath(os.path.join(plugins_dir, name + ".py"))
+        #print("loading custom", path)
+        model = load_custom_model(path)
+    elif name and name is not None and name.lower() != "none":
+        #print("loading standard", name)
+        model = _make_standard_model(name)
+    else:
+        model = None
+    _MODEL_CACHE[name] = model
```

Comment:
Is creating a _model_cache dict entry for None really a good idea? What does it mean?
It would always be ```_MODEL_CACHE[None] = None``` anyway.